pkg/endpoint/cache.go

```diff
@@ -113,6 +115,16 @@ func (ep *epInfoCache) GetIdentityLocked() identity.NumericIdentity {
 	return ep.identity
 }
 
+// GetEndpointNetnsCookie returns the network namespace cookie for the endpoint
+func (ep *epInfoCache) GetEndpointNetnsCookie() uint64 {
+	return ep.netNsCookie
+}
```

Comment:
The function body is same as `GetEndpointNetnsCookieLocked`. Is that expected? Should this function try to acquire a lock?

Reply:
This function is completely unused isn't it?

The fooLocked aspect is also a bit quirky.. my recollection is that part of the idea behind `epInfoCache` is that we take a snapshot of important info from the `Endpoint` object into a thread-local copy while holding a lock, then we can release the Endpoint lock and continue processing async. The fact that `epInfoCache` and the interfaces define `xxxLocked()` variants is a bit of a code smell, suggesting that knowledge of locks is leaking across interfaces when it shouldn't be.. but probably the epInfoCache needs to satisfy the same interfaces as `Endpoint` in different parts of the code. Ideally we'd work to remove those over time, but that's a broader effort so we don't need to address that in this PR. In context specifically of this PR change, if the data being accessed doesn't use a lock and it's only being referenced from this copy, then we probably don't need to name it with the `Locked` suffix.

Reply:
We need GetEndpointNetnsCookieLocked() in pkg/datapath/linux/config/config.go, so i had to implement this in other places to satisfy the interface.

Reply:
Took a look as well, and I don't understand why `linux/config/config.go` needs this?

Reply:
There is an existing implementation of [GetEndpointNetnsCookie()](https://github.com/cilium/cilium/blob/e541aa0d6bff185c3f45cacacb48830bc9b7527e/pkg/endpoint/identifiers.go#L166) but we need an a variant of this function which doesn't acquire the lock to be called from `linux/config/config.go`. I could just have `GetEndpointNetnsCookieLocked()` in cache.go and template.go, but created `GetEndpointNetnsCookie()` just to keep things consistent with other functions like `GetIdentity()` and `GetIdentityLocked()`. I can go ahead and remove `GetEndpointNetnsCookie()` from cache.go and template.go if that is more readable, but i don't think we can lose the Locked suffix.

Reply:
Hm I don't understand that whole function in the first place.
1. e.NetNsCookie is exported, so not all accesses are locked by definition.
2. It doesn't seem like the field is ever changed after creating the struct.

Maybe the right solution is to drop the lock (cc @aditighag why is it necessary?) or to turn the NetNsCookie into an `atomic` type.

Reply:
We can drop the `GetEndpointNetnsCookie` function altogether as `NetNsCookie`  is never updated after init, which is why it's exported similar to other exported ep fields. I'll send a PR.

Reply:
Thanks @aditinag. Do i wait for your changes and rebase ?. I could also create a commit to  drop GetEndpointNetnsCookie() from identifiers.go.

Reply:
This is the PR - https://github.com/cilium/cilium/pull/33856.

Reply:
I see that you have merged your PR. Updated the commit.

Its related because [Endpoint](https://github.com/cilium/cilium/blob/0a4ec831ec9c6b95b60fa2a73fe6e417149b442d/pkg/endpoint/endpoint.go#L144) needs to implement the [EndpointConfiguration](https://github.com/cilium/cilium/blob/0a4ec831ec9c6b95b60fa2a73fe6e417149b442d/pkg/datapath/types/config.go#L90) interface and we need a getter function for NetNs Cookie (which does not acquire any locks) in  linux/config/config.go.

Reply:
I'm confused: NetNscookie filed is exported. Why do you need a getter? See my above PR.

Reply:
The input to the function [writeStaticData]( https://github.com/cilium/cilium/blob/50b9a31dd25e5498079528bb534059106b5bbb51/pkg/datapath/linux/config/config.go#L985) is [EndpointConfiguration](https://github.com/cilium/cilium/blob/0a4ec831ec9c6b95b60fa2a73fe6e417149b442d/pkg/datapath/types/config.go#L90)  and we cannot access NetNscookie cookie directly. We call this function with [Endpoint](https://github.com/cilium/cilium/blob/0a4ec831ec9c6b95b60fa2a73fe6e417149b442d/pkg/endpoint/endpoint.go#L144) which implements the [EndpointConfiguration](https://github.com/cilium/cilium/blob/0a4ec831ec9c6b95b60fa2a73fe6e417149b442d/pkg/datapath/types/config.go#L90) interface.